rotkehlchen/tests/api/test_history_base_entry.py

```diff
@@ -635,3 +635,105 @@ def test_query_new_events(rotkehlchen_api_server_with_exchanges: 'APIServer') ->
+    db = DBHistoryEvents(rotki.data.db)
+    entries = [
+        {
+            'entry_type': 'asset movement event',
+            'timestamp': 1569924575000,
+            'balance': {'amount': '0.44', 'usd_value': '0'},
+            'event_type': 'deposit',
+            'location': 'bitfinex',
+            'unique_id': 'BITFINEX-543',
+            'asset': 'ETH',
+        }, {
+            'entry_type': 'asset movement event',
+            'timestamp': 1669924575000,
+            'balance': {'amount': '0.0569', 'usd_value': '0'},
+            'event_type': 'withdrawal',
+            'fee': '0.000004',
+            'location': 'bitfinex',
+            'unique_id': 'BITFINEX-344',
+            'asset': 'ETH',
+            'fee_asset': 'ETH',
+        },
+    ]
+    for entry in entries:
+        response = requests.put(
+            api_url_for(rotkehlchen_api_server, 'historyeventresource'),
+            json=entry,
+        )
+        result = assert_proper_sync_response_with_result(response)
+        assert 'identifier' in result
+        entry['identifier'] = result['identifier']
+
+    with rotki.data.db.conn.read_ctx() as cursor:
+        saved_events = db.get_history_events(
+            cursor=cursor,
+            filter_query=HistoryEventFilterQuery.make(),
+            has_premium=True,
+            group_by_event_ids=False,
+        )
+    assert len(saved_events) == 3  # including the fee event.
+
+    # test editing unknown fails
+    entry = entries[1].copy()
+    entry['identifier'] = 42
+    response = requests.patch(
+        api_url_for(rotkehlchen_api_server, 'historyeventresource'),
+        json=entry,
+    )
+    assert_error_response(
+        response=response,
+        contained_in_msg='Tried to edit event with id 42 but could not find it in the DB',
+        status_code=HTTPStatus.CONFLICT,
+    )
+    with rotki.data.db.conn.read_ctx() as cursor:
+        # now let's try to edit event_identifier for all possible events.
+        for idx, entry in enumerate(entries, start=1):
+            entry['identifier'] = idx
+            entry['event_identifier'] = f'new_eventid{idx}'
+            response = requests.patch(
+                api_url_for(rotkehlchen_api_server, 'historyeventresource'),
+                json=entry,
+            )
+            assert_simple_ok_response(response)
+
+        saved_events = db.get_history_events(
+            cursor=cursor,
+            filter_query=HistoryEventFilterQuery.make(),
+            has_premium=True,
+            group_by_event_ids=False,
+        )
+        assert len(saved_events) == 3
+        for idx, event in enumerate(saved_events):
+            serialized_event = event.serialize()
+            fields_to_exclude = {
+                'notes', 'extra_data', 'event_subtype', 'sequence_index',
+                'location_label', 'unique_id', 'event_identifier',
+                'fee', 'fee_asset', 'timestamp',
+            }
```

Comment:
nitpick: this can be outside the for loop

Reply:
+1. Do it.